catroid/src/main/java/org/catrobat/catroid/content/Sprite.java

```diff
@@ -664,4 +729,22 @@ public void updateUserVariableReferencesInUserVariableBricks(List<UserVariable>
 			}
 		}
 	}
+
+	public void setConvertToSingleSprite(boolean convertToSingleSprite) {
```

Comment:
I am not sure if you need that in other places, but if you only use that for cloning, maybe it would be easier and cleaner to have a method:
`public Sprite cloneAsSingleSprite() {
    this.convertToSingleSprite = true;
    Sprite clone = clone();
    this.convertToSingleSprite = false;
    return clone;
}`